sale_cancel_reason/tests/test_sale_cancel_reason.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013 Guewen Baconnier, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestSaleCancelReason(TransactionCase):
+    def test_sale_order_cancel_reason(self):
+        """
+        - Cancel a sales order with the wizard asking for the reason
+        - Then the sale order should be canceled and the reason stored
+        """
+        SaleOrderCancel = self.env["sale.order.cancel"]
+        wizard = SaleOrderCancel.create(
+            {"reason_id": self.reason.id, "order_id": self.sale_order.id}
+        )
+        wizard.with_context(
+            active_model="sale.order", active_ids=self.sale_order.id
+        ).action_cancel()
+        self.assertEqual(
+            self.sale_order.state, "cancel", "the sale order should be canceled"
+        )
+        self.assertEqual(self.sale_order.cancel_reason_id.id, self.reason.id)
+
+    def setUp(self):
```

Comment:
Move the setUp to setUpClass as it's way more performant since it runs only once.